BitwardenShared/UI/Auth/Login/TwoFactorAuth/TwoFactorAuthView.swift

```diff
@@ -120,11 +118,20 @@ struct TwoFactorAuthView: View {
         .scrollView(padding: 12)
     }
 
-    /// The image representing the email method
-    private var emailImage: some View {
-        Image(decorative: Asset.Images.Illustrations.emailOtp)
-            .resizable()
-            .frame(width: 124, height: 124)
+    /// The image representing the selected authentication method.
+    @ViewBuilder private var authMethodImage: some View {
+        switch store.state.authMethod {
+        case .email:
+            Image(decorative: Asset.Images.Illustrations.emailOtp)
+                .resizable()
+                .frame(width: 124, height: 124)
+        case .authenticatorApp:
+            Image(decorative: Asset.Images.Illustrations.authenticatorApp)
+                .resizable()
+                .frame(width: 124, height: 124)
+        default:
+            EmptyView()
+        }
     }
```

Comment:
🤔 I think it'd be nice to have a computed property in the state that just returns an optional image to show, then you can change this to return an `EmptyView` if the computed property is `nil` and bind the property to the image `decorative` if there's a value in there. Therefore if more images are added per authMethod you only need to change the state instead of creating one `Image` view per `authMethod`.
```suggestion
    /// The image representing the selected authentication method.
    @ViewBuilder private var authMethodImage: some View {
        if let authMethodImage = store.state.authMethodImage {
            Image(decorative: authMethodImage)
                .resizable()
                .frame(width: 124, height: 124)
        } else {
            EmptyView()
        }
    }
```
🤔 Additionally, I was wondering if it'd be good to replace this and the `detailText` with `PageHeaderView` as I think @KatherineInCode suggested a while back. Although, we'd need to make several things optional in the custom control and also it doesn't have the `detailImageAsset` so maybe it's not a good idea to replace it. Thoughts?

Reply:
Yeah, I don't think it would be good idea to change ```PageHeaderView``` for this case.
I think computed property is good approach here.